pennylane/transforms/condition.py

```diff
@@ -57,22 +64,24 @@ def __init__(
         super().__init__(wires=then_op.wires, do_queue=do_queue, id=id)
 
 
-def cond(measurement, then_op):
+def cond(condition, true_fn, false_fn=None):
     """Condition a quantum operation on the results of mid-circuit qubit measurements.
 
     Support for using :func:`~.cond` is device-dependent. If a device doesn't
     support mid-circuit measurements natively, then the QNode will apply the
     :func:`defer_measurements` transform.
 
     Args:
-        measurement (MeasurementValue): a measurement value to consider, for
-            example the output of calling :func:`~.measure`
-        then_op (Operation): The PennyLane operation to apply if the condition
-            applies.
+        condition (MeasurementValue[bool]): a conditional expression involving a mid-circuit
+           measurement value (see :func:`.pennylane.measure`)
+        true_fn (callable): The quantum function of PennyLane operation to
+            apply if ``condition`` is ``True``
+        false_fn (callable): The quantum function of PennyLane operation to
+            apply if ``condition`` is ``False``
 
     Returns:
-        function: A new function that applies the conditional equivalent of ``then_op``. The returned
-        function takes the same input arguments as ``then_op``.
+        function: A new function that applies the conditional equivalent of ``true_fn``. The returned
+        function takes the same input arguments as ``true_fn``.
 
     **Example**

```

Comment:
@antalszava the example here is not up to our usual standards for docstring examples (e.g., it doesn't contain any text explaining the context/code, and showing outputs).

It also seems to be missing:

- examples with qfuncs,
- examples with then/else, and
- examples showing how to deal with conditional functions with different signatures